src/itsdangerous/serializer.py

```diff
@@ -135,7 +174,9 @@ def secret_key(self) -> bytes:
         """
         return self.secret_keys[-1]
 
-    def load_payload(self, payload: bytes, serializer: t.Any | None = None) -> t.Any:
+    def load_payload(
+        self, payload: bytes, serializer: _PDataSerializer[_TAnyStr] | None = None
```

Comment:
```suggestion
        self, payload: bytes, serializer: _PDataSerializer[t.Any] | None = None
```
The type of the passed in serializer doesn't matter, since it is re-checked.